internal/wrphandlers/xmidt_agent_crud/handler.go

```diff
@@ -36,31 +37,35 @@ func New(egress wrpkit.Handler, source string, logLevel loglevel.LogLevel) (*Han
 }
 
 func (h *Handler) HandleWrp(msg wrp.Message) error {
+	response := msg
+	response.Destination = msg.Source
+	response.Source = h.source
+	response.ContentType = "application/json"
 	payload := make(map[string]string)
 
 	err := json.Unmarshal(msg.Payload, &payload)
-	if err != nil { // do we still want to return a response here?
-		return err // this would be sent to the eventor error handler
+	if err != nil {
+		statusCode := int64(http.StatusInternalServerError)
+		response.Status = &statusCode
+		response.Payload = []byte(fmt.Sprintf(`{statusCode: %d, message: "%s"}`, statusCode, err.Error()))
+		return h.egress.HandleWrp(response)
```

Comment:
I think we should response back with error otherwise it'll make trouble shooting harder than it should be